### PR TITLE
#187419015 Admin should be able to acivate/reactivate an account for various reasons.

### PR DESCRIPTION
**Pull request by YvetteNyibuka**
## What does this PR do?
Enabling/Disabling User account by admin
## Description of Task to be completed?
- Enabling user account to make it active
- Disabling user account in case he/she violates `App's terms and conditions`
## How should this be manually tested?
- Clone this repository, go to the branch `ft-accountEnable-disable-187419015`
- Have `.env` file with proper variables
- install all packages using `npm install`
- run `npm run dev` 
- login as admin and try eabling/disabling account

## What are the relevant pivotal tracker/Trello stories?
#187419015

### CODE REVIEW

File: src/redux/features/EnableAccountSlice.ts [hunk 1]
```diff
@@ -0,0 +1,78 @@
+/* eslint-disable @typescript-eslint/no-explicit-any */
```
Comment (Angemichel12) [inline src/redux/features/EnableAccountSlice.ts]: Remove this line to disable eslint

File: src/redux/features/EnableAccountSlice.ts [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+				reason,
+			});
+			return data;
+		} catch (error) {
+			return rejectWithValue((error as DynamicData).response);
+		}
+	},
+);
+
+type EnableAccountState = {
```
Comment (Angemichel12) [inline src/redux/features/EnableAccountSlice.ts]: Move this type to types directory, for me it is good practice.

File: .vscode/extensions.json [hunk 3]
```diff
@@ -0,0 +1,3 @@
+{
+	"recommendations": ["github.copilot"]
```
Comment (Angemichel12) [inline .vscode/extensions.json]: why are you recommended us to use github.copilot

**Review by Angemichel12 — APPROVED** [review]

**Review by GarrixA — APPROVED** [review]

File: .vscode/extensions.json (hunk 3 above)
Comment (leandreAlly) [reply .vscode/extensions.json]: @Angemichel12 thanks for elaborating this issue. @YvetteNyibuka delete this ASAP, we don't need your AI configurations.

File: src/components/Forms/editUserForm.tsx [hunk 4]
```diff
@@ -30,20 +41,24 @@ const EditUserForm = (props: userFormType) => {
...
 		}
-	}, [dispatch, roles]);
+		dispatch(setEnable(isAccountActive ? 'true' : 'false'));
+	}, [dispatch, isAccountActive]);
 
 	const handleNavigation = () => {
-		if (url) {
-			navigate(-1);
-		}
+		navigate(-1);
```
Comment (leandreAlly) [inline src/components/Forms/editUserForm.tsx:61]: what are you real trying to here? navigate(-1); ? provide more clarifications on this code?

File: src/components/adminDashboard/getUser.tsx [hunk 5]
```diff
@@ -29,16 +30,22 @@ const GetUser = (props: getUserType) => {
 	const dispatch = useAppDispatch();
 	const itemsPerPage = 8;
 
+	console.log('dataaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa', roles);
+
```
Comment (leandreAlly) [inline src/components/adminDashboard/getUser.tsx]: remove this console.log()

**Review by leandreAlly — CHANGES_REQUESTED** [review]

**Review by leandreAlly — APPROVED** [review]

**Pull request merged**